### PR TITLE
Rename `Value` to `Immediate` for miri

### PR DESCRIPTION
**Pull request by oli-obk**
r? @RalfJung

### CODE REVIEW

File: src/librustc_mir/interpret/operand.rs [hunk 1]
```diff
@@ -146,32 +146,32 @@ impl<'tcx, Tag> ScalarMaybeUndef<Tag> {
 /// In particular, thanks to `ScalarPair`, arithmetic operations and casts can be entirely
```
Comment (RalfJung) [inline src/librustc_mir/interpret/operand.rs]: The doc comment still refers to the old name.

**Review by RalfJung — CHANGES_REQUESTED** [review]
r=me with the nit fixed.

Thanks!

File: src/librustc_mir/interpret/operand.rs [hunk 2]
```diff
@@ -146,32 +146,32 @@ impl<'tcx, Tag> ScalarMaybeUndef<Tag> {
...
 /// defined on `Value`, and do not have to work with a `Place`.
```
Comment (RalfJung) [inline src/librustc_mir/interpret/operand.rs]:
```suggestion
/// defined on `Immediate`, and do not have to work with a `Place`.
```

**Comment by RalfJung** [comment]
LGTM. :) Needs a rebase though.

**Comment by oli-obk** [comment]
done

**Pull request merged**